### DISCUSSION TITLE
Is there a way to execute terraform apply on a schedule using tfaction?

### DISCUSSION THREAD

**Discussion by 5st7** [suzuki-shunsuke/tfaction#1795] [Deprecated (Question and Answer)] [answered]
### Question

Thank you for providing such a wonderful OSS.

I would like to execute terraform apply at a specific time using Cron to modify the resources based on the time of day or to correct drifts. I'm considering a setup where GitHub Actions creates a PR to modify the appropriate resources, automatically merges it, and then applies the changes. However, is there a better approach using tfaction?

### Background of the question

 background:
- modify resources only on holidays to reduce costs.
- reduce drift.

For these reasons, I’m considering executing terraform apply on a schedule.

**Comment by suzuki-shunsuke** [comment, marked as answer]
> is there a better approach using tfaction?

tfaction doesn't provide a way to run `terraform apply` periodically.
It's potentially dangerous, so we have no plan to support it.

Probably you are aware, but tfaction supports [the drift detection](https://suzuki-shunsuke.github.io/tfaction/docs/feature/drift-detection) instead.

> GitHub Actions creates a PR to modify the appropriate resources, automatically merges it, and then applies the changes.

I think this would work well.

If you don't want to create prs, you can implement the workflow based on [apply.yaml](https://github.com/suzuki-shunsuke/tfaction-example/blob/main/.github/workflows/apply.yaml) or [schedule-detect-drifts.yaml](https://github.com/suzuki-shunsuke/tfaction-example/blob/main/.github/workflows/schedule-detect-drifts.yaml).
[tfaction/terraform-apply](https://github.com/suzuki-shunsuke/tfaction/tree/main/terraform-apply) isn't available because it requires a plan file in GitHub Artifacts, but you can still utilize other actions such as [tfaction/setup](https://github.com/suzuki-shunsuke/tfaction/blob/main/setup/action.yaml).
